lib/killbill_client/models/transaction.rb

```diff
@@ -138,6 +138,59 @@ def cancel_scheduled_payment(user = nil, reason = nil, comment = nil, options =
                           }.merge(options)
       end
 
+      def chargerback_by_external_key(payment_external_key, amount, currency, effective_date = nil, user = nil, reason = nil, comment = nil, options = {}, refresh_options = nil)
+        follow_location = delete_follow_location(options)
+        refresh_payment_with_failure_handling(follow_location, refresh_options || options) do
+        payload                          = PaymentTransactionAttributes.new
+        payload.payment_external_key     = payment_external_key
```

Comment:
No need to recreate a `PaymentTransactionAttributes` object, since `chargeback_by_external_key` is already an instance method: the caller would populate the fields on the object. See how these methods are used here for instance: https://github.com/killbill/killbill-admin-ui/blob/8e9540a019b6c9595ba95463bd4d6365ba09e1ee/app/controllers/kaui/chargebacks_controller.rb#L15

Also, all of these methods should be symmetric in their implementation and signature, only the name really would differ (we can keep `def XXX` for operations by `id` and `def XXX_by_external_key` for operations by `external_key`).